nan.h

```diff
@@ -909,6 +924,32 @@ class NanEscapableScope {
     return function_template->HasInstance(value);
   }
 
+  namespace Nan { namespace imp {
+  NAN_INLINE void
+  widenString(std::vector<uint16_t> *ws, const uint8_t *s, int l = -1) {
+    size_t len = static_cast<size_t>(l);
+    if (l < 0) {
+      len = strlen(reinterpret_cast<const char*>(s));
+    }
+    assert(len <= INT_MAX && "string too long");
+    ws->resize(len);
+    std::copy(s, s + len, ws->begin());
+  }
+  }  // end of namespace imp
+  }  // end of namespace Nan
+
+  NAN_INLINE NanMaybeLocal<v8::String>
+  NanNewOneByteString(const uint8_t * value, int length = -1) {
+    std::vector<uint16_t> wideString;
+    Nan::imp::widenString(&wideString, value, length);
+    if (wideString.size() == 0) {
+      return Factory<v8::String>::return_t(v8::String::Empty());
```

Comment:
Just a note: you don't have to return String::Empty() yourself, V8 already does that.